llm/cli.py

```diff
@@ -443,35 +444,185 @@ def chat(
         for chunk in response:
             print(chunk, end="")
             sys.stdout.flush()
-        response.log_to_db(db)
+        parent_id = get_parent_id(response, id)
+        response.log_to_db(db, parent_id)
         print("")
 
 
+def get_most_recent_active_conversation(db):
+    return next(db.query("""
+        SELECT conversation_id, MAX(datetime_utc) as last_active
+        FROM responses
+        GROUP BY conversation_id
+        ORDER BY last_active DESC
+        LIMIT 1
+    """), {}).get("conversation_id")
+
+
+def get_head(db):
+    try:
+        return db['state'].get('head')['value']
+    except sqlite_utils.db.NotFoundError:
+        return None
+
+
+def get_response_parent(response, db):
+    if response.parent_id:
+        # Use explicit parent if set
+        return next(db.query("SELECT * FROM responses WHERE id = ?", [response.parent_id]), None)
+
+    if not response.conversation:
+        return {'id': None}
+
+    # Fall back to previous response in conversation
```

Comment:
get_response_parent will fall back to the default behaviour, so it will work like normal with existing dbs without having to populate the parent_id for existing responses.